syntax = "proto3";

package rdv.v1alpha;

import "google/protobuf/timestamp.proto";

message RDV {
  string id = 1;
  string name = 2;
  string idUser = 3;
  string participantId = 4;

  google.protobuf.Timestamp created_at = 5;
  google.protobuf.Timestamp updated_at = 6;
  google.protobuf.Timestamp rdv_date = 7;

  string rdv_hour = 8;

  string idLieu = 9;
}

service RdvCRUDService {
  rpc Get (GetRequest) returns (GetResponse);
  rpc Add (AddRequest) returns (AddResponse);
  rpc Update (UpdateRequest) returns (UpdateResponse);
  rpc Delete (DeleteRequest) returns (DeleteResponse);
}

message GetRequest {
  string id = 1;
  string name = 2;
  string idUser = 3;
  string participantId = 4;
}

message GetResponse {
  repeated RDV rdv = 1;
}

message AddRequest {
  string name = 1;
  string idUser = 2;
  string participantId = 3;
}

message AddResponse {
  RDV rdv = 1;
}

message UpdateRequest {
  RDV rdv = 1;
}

message UpdateResponse {
  RDV rdv = 1;
}

message DeleteResponse {
  RDV rdv = 1;
}

message DeleteRequest {
  string id = 1;
}